### PR TITLE
Docker environment

### LINKED ISSUE #44: Typo in frontend/src/app/signup/signup.component.ts
Hi, there's a typo in:
frontend/src/app/signup/signup.component.ts
Where is:
import { Subject } from 'rxjs/SUbject';
Should be:
import { Subject } from 'rxjs/Subject';

### PR DESCRIPTION
**Pull request by apeixinho**
I've implemented a docker environment with your project.
I don't know if a pull request to master is the best approach or perhaps a new branch just for docker environment.
I've also upgraded some versions in package.json and pom.xml. 
Everything seems to be working as usual.
Also updated the main README documentation with docker instructions and created a table of contents.
Any doubt feel free to ask or if I missed something let me know. 

#Closes #44

### CODE REVIEW

**Comment by bfwg** [comment]
@apeixinho Please Let me know if you need any help.

**Comment by apeixinho** [comment]
Thanks @bfwg . Just quite busy, lately. I've seen your reviews and I'll implement the necessary changes. Some stuff I just forgot like: the proxy name on docker is different, etc.
Perhaps, this weekend... I'll keep in touch.

**Comment by apeixinho** [comment]
I forgot to ask ... What do you think of a docker branch? Is it more manageable then putting docker implementation on master?

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+# Define the services/containers to be run
+services:
+  angular: # name of the first service
+    build: frontend # specify the directory of the Dockerfile
+    # volumes:
+    # - .:/home/app
+    # - /home/app/node_modules
+    ports:
+    - "80:4200"
```
[the remark this answers is not in the document]
Comment (apeixinho) [reply docker-compose.yml]: Will be changed accordingly to 4200:4200

File: README.md [hunk 2]
```diff
@@ -16,17 +16,85 @@
...
+Successfully tagged angularspringstarter_angular:latest
+Creating angularspringstarter_springboot_1 ... done
+Creating angularspringstarter_angular_1    ... done
+
+# the app will be available at localhost
+# if you wish to see your newly created containers issue:
+docker ps
+
+# if you wish to have access to a running shell inside a running container you can issue:
+docker exec -it <container name> sh
```
[the remark this answers is not in the document]
Comment (apeixinho) [reply README.md]: Will be updated accordingly

File: server/src/main/java/com/bfwg/security/auth/AnonAuthentication.java [hunk 3]
```diff
@@ -26,26 +26,28 @@ public Object getPrincipal() {
...
+//            return true;
+//        }
+//        if ( obj == null ) {
+//            return false;
+//        }
+//        if ( getClass() != obj.getClass() ) {
+//            return false;
+//        }
+//        return true;
+//    }
```
[the remark this answers is not in the document]
Comment (apeixinho) [reply server/src/main/java/com/bfwg/security/auth/AnonAuthentication.java]: Will be removed accordingly

File: server/src/main/java/com/bfwg/security/auth/TokenAuthenticationFilter.java [hunk 4]
```diff
@@ -1,8 +1,8 @@
 package com.bfwg.security.auth;
 
 import com.bfwg.security.TokenHelper;
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
+//import org.apache.commons.logging.Log;
+//import org.apache.commons.logging.LogFactory;
 import org.springframework.beans.factory.annotation.Autowired;
```
[the remark this answers is not in the document]
Comment (apeixinho) [reply server/src/main/java/com/bfwg/security/auth/TokenAuthenticationFilter.java]: Will be removed accordingly

File: frontend/proxy.conf.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "/api": {
-    "target": "http://localhost:8080",
+    "target": "http://springboot:8080",
```
[the remark this answers is not in the document]
Comment (apeixinho) [reply frontend/proxy.conf.json]: will be reverted back to localhost

[the remark this answers is not in the document]
Comment (apeixinho) [reply frontend/proxy.conf.json]: Workaround for this doesn't seem to be working: tried with 0.0.0.0:8080 and the original localhost:8080 still not working on docker.
[HPM] Error occurred while trying to proxy request /api/refresh from localhost:4200 to http://0.0.0.0:8080 (ECONNREFUSED)

**Comment by apeixinho** [comment]
@bfwg This perhaps just seems to much trouble. Better to keep the master branch how it is. My advice is not accept the pull request and just leave things how they were previously

**Comment by bfwg** [comment]
@apeixinho Thanks for the quick response. I can create a `docker` branch for you, and you can point this PR against that branch if you want. Maybe also remove the "start without docker" section in the README.md. Let me know what you think. 😄

**Comment by apeixinho** [comment]
Sorry for the late response. 
Agreed, a separate branch for docker environment seems more manageable. 
I'll try and take a look this weekend.
I'll, also, later take a look at an approach to testing with docker.

**Comment by bfwg** [comment]
No worries, I've reopened the PR. Feel free to make changes and point the branch to the `docker` branch.

**Pull request closed without merging**